src/components/flow/ResultsAsSimulation.tsx

```diff
@@ -1,24 +1,446 @@
+	);
+
+	const [crossroad, setCrossroad] = useState<ResponseCrossroad | null>(null);
+	const [crossroadImage, setCrossroadImage] = useState<string | undefined>(undefined);
+	const [exitEntrancePoints, setExitEntrancePoints] = useState<ExitEntrancePoint[]>(
+		[],
+	);
+	const [connections, setConnections] = useState<ResponseConnection[]>([]);
+	const [trafficLights, setTrafficLights] = useState<TrafficLight[]>([]);
+	const [collisions, setCollisions] = useState<ResponseCollision[]>([]);
+
+	const [showFailureAlert, setShowFailureAlert] = useState(false);
+	const [failureMessage, setFailureMessage] = useState("");
+
+	useEffect(() => {
+		axios
+			.get<CrossroadDescriptionResponse>(`/crossroad/${crossroadId}`, {
+				headers: {
+					Authorization: `Bearer ${
+						loggedUser !== null ? loggedUser.jwtToken : getUserJWTToken()
+					}`,
+				},
+			})
+			.then((response) => {
+				const crossingsData: CrossroadDescriptionResponse = response.data;
+				setCrossroad(crossingsData.crossroad);
+				setExitEntrancePoints(crossingsData.roads);
+				setConnections(crossingsData.connections);
+				setTrafficLights(crossingsData.trafficLights);
+				setCollisions(crossingsData.collisions);
+				setCrossroadImage(crossingsData.image);
+			})
+			.catch((error) => {
+				console.error(error);
+				setFailureMessage(
+					`Request failed: ${error.code}!\nTry going back to crossroads-list and viewing the model again`,
+				);
+				setShowFailureAlert(true);
+			});
+	}, []);
+
+	const getLightSequence = (lightIdx: number, lightsSeq: number[][]) => {
+		const newIdx = lightIdx - 1;
+		if (newIdx >= 0 && newIdx < lightsSeq.length) {
+			return lightsSeq[newIdx];
+		} else {
+			return [];
+		}
+	};
+
+	const [running, setRunning] = useState(false);
+
+	const [timer, setTimer] = useState(0);
+	const timeInterval = 60;
+	const timeDelta = 200;
+	const lightColors = [LightColors.RED, LightColors.GREEN, LightColors.YELLOW];
+	const [lights, setLights] = useState<Map<number, LightColors>>(
+		new Map([
+			[1, LightColors.YELLOW],
+			[2, LightColors.YELLOW],
+			[3, LightColors.YELLOW],
+			[4, LightColors.YELLOW],
+		]),
+	);
+
+	const [cars, setCars] = useState<Map<number, number>>(
+		new Map([
+			[1, 0],
+			[2, 0],
+		]),
+	);
+
+	function howManyCarsArrived(flow: number) {
+		let n = 0;
+		while (flow > 0 && Math.random() < flow) {
+			n += 1;
+		}
+		return n;
+	}
+
+	function getConnectionsIdsForRoadIdx(index: number) {
+		// console.log("conRoad", conRoad);
+		const connections: number[] = [];
+		conRoad.forEach(function (value, key) {
+			if (value == index) {
+				connections.push(key);
+			}
+		});
+		return connections;
+	}
+
+	function getWhichConnectionToGo(connectionsIds: number[]) {
+		let chanceFloor = 0;
+		let chanceCeil = 0;
+		const randomNumber = Math.random();
+		let chosenConnection = 0;
+		// console.log("connectionsIds", connectionsIds);
+		// console.log("randomNumber", randomNumber);
+		// console.log("conChances", conChances);
```

Comment:
Jeśli to już jest finalnie done to wszystkie console.logi zakomentowanie i nie trzeba wyjebać

Reply:
dobra, wywale i najwyzej nowe sb dodam xd